crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1713,7 +1713,12 @@ pub async fn list_customer_payment_method(
         };
         customer_pms.push(pma.to_owned());
 
-        let redis_conn = state.store.get_redis_conn();
+        let redis_conn = state
+            .store
+            .get_redis_conn()
+            .change_context(errors::ApiErrorResponse::InternalServerError)
```

Comment:
Can you try to use `ErrorSwitch` instead of using `change_context` everywhere?

For your reference 
https://github.com/juspay/hyperswitch/blob/71b75c653845685b71c6fb6007a718b6cb2c65c5/crates/router/src/connector/utils.rs#L897

Reply:
@dracarys18  is that something you expect to be done for `DummyConnectorErrors` ?

Reply:
Let's take this up change on a separate PR, since this one has been open for quite some time.